### PR TITLE
[Enhancement]Added a metric for geo replication for tracking replicated subscriptions snapshot timeouts

### LINKED ISSUE #21793: Add a metric for geo replication for tracking replicated subscriptions snapshot timeouts
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Motivation

Geo replication replicated subscriptions (PIP-33) [snapshot creation might time out](https://github.com/apache/pulsar/wiki/PIP-33%3A-Replicated-subscriptions#constructing-a-cursor-snapshot).
The code contains a debug log message when this happens: https://github.com/apache/pulsar/blob/465fac523da946553b09298e13dc7dfcecfb6c78/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/ReplicatedSubscriptionsController.java#L256
When this happens, the subscription state won't be reflected on the remote side and a backlog would build up.
There's no metric to detect this situation. 

### Solution

Add a new metric `pulsar_replicated_subscriptions_snapshot_timeouts` which is a counter (that only resets when the broker restarts).

### Alternatives

_No response_

### Anything else?

Increasing the timeout threshold `replicatedSubscriptionsSnapshotTimeoutSeconds=30` -> `replicatedSubscriptionsSnapshotTimeoutSeconds=60` could help resolve the situation. This metric would help detect when it would be necessary.

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by nikam14**
Fixes #21793

### Motivation
Geo replication replicated subscriptions (PIP-33) [snapshot creation might time out](https://github.com/apache/pulsar/wiki/PIP-33%3A-Replicated-subscriptions#constructing-a-cursor-snapshot).
The code contains a debug log message when this happens
When this happens, the subscription state won't be reflected on the remote side and a backlog would build up.
There's no metric to detect this situation.

### Modifications
Add a new metric `pulsar_replicated_subscriptions_snapshot_timeouts` which is a counter (that only resets when the broker restarts).

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [x] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [x] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

**Comment by dao-jun** [comment]
1. it's better to use OpenTelemetry
2. we need a proposal

**Comment by lhotari** [comment]
> 1. it's better to use OpenTelemetry
> 2. we need a proposal

@dao-jun we don't have Otel in use yet. Yes, we can handle this in a proposal.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/ReplicatedSubscriptionsController.java [hunk 1]
```diff
@@ -75,6 +76,10 @@ public class ReplicatedSubscriptionsController implements AutoCloseable, Topic.P
                     "Counter of currently pending snapshots")
             .register();
 
+    private static final Counter timedoutSnapshotsMetric = Counter
+            .build().name("pulsar_replicated_subscriptions_snapshot_timeouts")
```
Comment (lhotari) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/ReplicatedSubscriptionsController.java:80]: Perhaps `pulsar_replicated_subscriptions_timedout_snapshots` would be more consistent with the existing `pulsar_replicated_subscriptions_pending_snapshots`? I don't have a strong opinion about this.

**Review by lhotari — APPROVED** [review]
LGTM. I think that the lack of this metric could be considered a significant problem is replicated subscription observability and should be added to LTS version.

**Comment by lhotari** [comment]
Thanks for the contribution @nikam14 !

**Review by codelipenghui — COMMENTED** [review]
Could you please help add an unit test to avoid the regression?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/ReplicatedSubscriptionsController.java [hunk 2]
```diff
@@ -75,6 +76,10 @@ public class ReplicatedSubscriptionsController implements AutoCloseable, Topic.P
                     "Counter of currently pending snapshots")
             .register();
 
+    private static final Counter timedoutSnapshotsMetric = Counter
```
Comment (asafm) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/ReplicatedSubscriptionsController.java:79]: I don't understand how this helps you understand the severity of the issue? It's basically serves a signal that something bad happened. Don't you need to know the backlog size, or am missing something on how this snapshotting process works?

**Comment by asafm** [comment]
FYI @dragosvictor

**Review by lhotari — APPROVED** [review]

**Pull request merged**